glotaran/plugin_system/test/test_data_io_registration.py

```diff
@@ -25,17 +25,19 @@
 from glotaran.plugin_system.data_io_registration import save_dataset
 from glotaran.plugin_system.data_io_registration import set_data_plugin
 from glotaran.plugin_system.data_io_registration import show_data_io_method_help
+from glotaran.plugin_system.data_io_registration import supported_file_extensions_data_io
+from glotaran.testing.plugin_system import monkeypatch_plugin_registry_data_io
 
 if TYPE_CHECKING:
+    from collections.abc import Sequence
     from typing import Any
 
     from _pytest.capture import CaptureFixture
-    from _pytest.monkeypatch import MonkeyPatch
 
     from glotaran.typing import StrOrPath
 
 
-class MockDataIO(DataIoInterface):
+class MockDataIo(DataIoInterface):
```

Comment:
I understand you have adopted the "io" is a word hence the capitalization is Io instead of IO, but widely adopted convention states the following: 

> When using acronyms, use Pascal case or camel case for acronyms more than two characters long. For example, use HtmlButton or htmlButton. However, you should capitalize acronyms that consist of only two characters, such as System.IO instead of System.Io.
Ref: https://learn.microsoft.com/en-us/previous-versions/dotnet/netframework-1.1/141e06ef(v=vs.71)?redirectedfrom=MSDN

See also discussion here: https://stackoverflow.com/a/15526526

Reply:
Not sure if this applies here since the `Io` is not stand alone. 🤔
Also, Python naming conventions aren't the same as `dotnet`.
In any case, this change makes it consistent with `DataIoInterface`, where a change to `DataIOInterface` would make it harder to read.